### ISSUE TITLE
Calling parent method doesn't work

### ISSUE THREAD

**Issue by drnextgis** [openlayers/ol3#5062]
Trying to call parent method accordingly API [docs](http://openlayers.org/en/v3.14.2/apidoc/ol.html#.inherits) but it doesn't work:

```
var Person = function(name) {
  this.name = name;
}

var Employee = function(name, position) {
    Employee.base(this, name);
    this.position = position;
}

ol.inherits(Employee, Person);

emp = new Employee('denis', 'webdev');
VM4311:7 Uncaught TypeError: Employee.base is not a function(…)
```

**Comment by nicholas-l** [comment]
I don't think base should be called in the instantiation. I think call should be used like below although I havent tested this:

```
var Person = function(name) {
  this.name = name;
}

Person.prototype.updateName = function(name) {
  this.name = name;
}

var Employee = function(name, position) {
    Employee.call(this, name);
    this.position = position;
}
 Employee.prototype.updateName = function(name) {
  Person.base(this, 'updateName', 'Employee:' + name);
}

ol.inherits(Employee, Person);

emp = new Employee('denis', 'webdev');
```

Employee instantiates the new instance with a call (Person.call) to Person then updateName calls, using base, the updateName of the Person class.

**Comment by drnextgis** [comment]
@nicholas-l there are some errors in your example: 

1. `Person.call` should be used instead `Employee.call`
2. `Employee.prototype.updateName` should be placed after `ol.inherits` because  `ol.inherits` function overwrites the child's prototype by an instance of the parent
3. `Employee.base` should be used instead `Person.base` (accordingly docs)

But anyway it doesn't work.

**Comment by nicholas-l** [comment]
OOps yes, you are correct on all three counts :) so more like this:

```
var Person = function(name) {
  this.name = name;
}

Person.prototype.updateName = function(name) {
  this.name = name;
}

var Employee = function(name, position) {
    Person.call(this, name);
    this.position = position;
}
ol.inherits(Employee, Person);

Employee.prototype.updateName = function(name) {
  Employee.base(this, 'updateName', 'Employee:' + name);
}

emp = new Employee('denis', 'webdev');
```

Looks like this runs in ol-debug.js but not in ol.js

**Comment by nicholas-l** [comment]
Could [this](https://github.com/openlayers/ol3/blob/master/externs/readme.md#override-methods-in-custom-classes) be something to do with it?

**Comment by drnextgis** [comment]
Exactly. Is it expected behaviour that `base` function mentioned in API docs but doesn't work with `ol.js`?

**Comment by ahocevar** [comment]
Sorry for the confusion. The wiki page linked above is not relevant. This is a documentation problem. The `base` method is definitely not available. You have to know the superclass yourself.

Personally, I'd recommend to use class/inheritance tools from someone else, not `ol.inherits` (which is just an alias for `goog.inherits`, and only works well when you build your application together with OpenLayers using the Google Closure library).

That said, a pull reqeust fixing the documentation would be much appreciated.

**Comment by drnextgis** [comment]
PR https://github.com/openlayers/ol3/pull/5063

**Comment by ahocevar** [comment]
Documentation fixed with #5063.
